Adds 'spo site apppermission add' command. Closes #2208 (#2296)
pnp/cli-microsoft365: docs/docs/cmd/spo/site/site-apppermission-add.md

# spo site apppermission add

Adds a specific application permissions to the site

## Usage

```sh
m365 spo site apppermission add [options]
```

## Options

`-u, --siteUrl <siteUrl>`
: URL of the site collection to add the permission

`-p, --permission <permission>`
: Permission to site (`read`, `write`, `read,write`). If multiple permissions have to be granted, they have to be comma separated ex. `read,write`

> `-i, --appId <appId>`

Reviewer: Since `appId` is optional, it should be denoted as `[appId]`

> : App Id

Reviewer: ```suggestion
: Client ID of the Azure AD app for which to grant permissions
```

> `-n, --appDisplayName <appDisplayName>`

Reviewer: Since `appDisplayName` is optional, it should be denoted as `[appDisplayName]`

> : App display name

Reviewer: ```suggestion
: Display name of the Azure AD app for which to grant permissions
```

--8<-- "docs/cmd/_global.md"

## Remarks

> Pass in appId and/or addDisplayName, use both for best performance avoiding the extra lookup.

Reviewer: ```suggestion

To set permissions, specify at minimum either `appId` or `addDisplayName`. For best performance specify both values to avoid extra lookup.
```

## Example

> Adds a specific application permission to the _https://contoso.sharepoint.com/sites/project-x_ site collection with _read_ permission 

Reviewer: ```suggestion
Grants the specified app the _read_ permission to site _https://contoso.sharepoint.com/sites/project-x_
```
